File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, h: List[int]) -> int:
        l, r = 0, len(h) - 1
        mL, mR = h[l], h[r]
        c = 0

        while l < r:
            if mL < mR:
                l += 1
                mL = max(mL, h[l])
                c += mL - h[l]
            else:
                r -= 1
                mR = max(mR, h[r])
                c += mR - h[r]

        return c

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
   public:
    int lengthOfLongestSubstring(string s) {
        if (s.empty()) return 0;

        unordered_set<char> window;
        int L = 0;
        int length = INT_MIN;

        for (int R = 0; R < s.size(); R++) {
            while (window.count(s[R]) > 0) {
                window.erase(s[L]);
                L++;
            }
            window.insert(s[R]);
            length = max(length, (int)window.size());
        }

        return length;
    }
};
